tests/src/test/java/edu/snu/onyx/tests/runtime/master/scheduler/SingleTaskGroupQueueTest.java

```diff
@@ -110,6 +112,8 @@ public void testPushPriority() throws Exception {
 
     final CountDownLatch countDownLatch = new CountDownLatch(2);
 
+    final AtomicBoolean passed = new AtomicBoolean(true);
+
     // This mimics Batch Scheduler's behavior
     executorService.execute(() -> {
```

Comment:
Instead of `AtomicBoolean passed`, can you use `Callable` with a timeout?

```java
@timeout
void test throws Exception {
  futureList = submitToExecutorService(callableList)
  
  // throws ExecutionException if the callable throws an error 
  // indefinite blocking is caught by the timeout
  futureList.forEach(future -> future.get());
}
```

Ditto on `testPullPriority()`, `testPushRemoveAndAddStageDependency()`, and `testContainerTypeAwareness()` to fail the tests on errors.

Reply:
I'm actually working on this in #683 :)